xclim/core/calendar.py

```diff
@@ -807,3 +807,168 @@ def time_bnds(group, freq):
+    )
+
+    if start is not None:
+        mm, dd = map(int, start.split("-"))
+        starts = xr.apply_ufunc(
+            lambda y: datetime_classes[calendar](y, mm, dd),
+            base.dt.year,
+            vectorize=True,
+        )
+        start_doy = starts.dt.dayofyear
+        start_doy = start_doy.where(start_doy > base_doy, start_doy + doy_max)
+    else:
+        start_doy = base_doy
+
+    return base_doy, start_doy, doy_max
+
+
+def doy_to_days_since(
+    da: xr.DataArray,
+    start: Optional[DayOfYearStr] = None,
+    calendar: Optional[str] = None,
+):
+    """Convert day-of-year data to days since a given date
+
+    This is useful for computing meaningful statistics on doy data.
+
+    Parameters
+    ----------
+    da: xr.DataArray
+      Array of "day-of-year", usually int dtype, must have a `time` dimension.
+      Sampling frequency should be finer or similar to yearly and coarser then daily.
+    start: date of year str, optional
+      A date in "MM-DD" format, the base day of the new array.
+      If None (default), the `time` axis is used.
+      Passing `start` only makes sense if `da` has a yearly sampling frequency.
+    calendar: str, optional
+      The calendar to use when computing the new interval.
+      If None (default), the calendar of the `time` axis is used.
+      All time coordinates of `da` must exist in this calendar.
+      No check is done to ensure doy values exist in this calendar.
+
+    Returns
+    -------
+    xr.DataArray
+      Same shape as `da`, int dtype, day-of-year data translated to a number of days since a given date.
+      If start is not None, there might be negative values.
+
+    Notes
+    -----
+    The time coordinates of `da` are considered as the START of the period. For example, a doy value of
+    350 with a timestamp of '2020-12-31' is understood as '2021-12-16' (the 350th day of 2021).
+    Passing `start=None`, will use the time coordinate as the base, so in this case the converted value
+    will be 350 "days since time coordinate".
+
+    Examples
+    --------
+    >>> time = date_range('2020-07-01', '2021-07-01', freq='AS-JUL')
+    >>> da = xr.DataArray([190, 2], dims=('time',), coords={'time': time})  # July 8th 2020 and Jan 2nd 2022
+    >>> doy_to_days_since(da, start='10-02').values  # Convert to days since Oct. 2nd, of the data's year.
+    array([-86, 92])
+    """
+    base_calendar = get_calendar(da)
+    calendar = calendar or base_calendar
+    dac = convert_calendar(da, calendar)
+
+    base_doy, start_doy, doy_max = _doy_days_since_doys(dac.time, start)
+
+    # 2cases:
+    # val is a day in the same year as its index : da - offset
+    # val is a day in the next year : da + doy_max - offset
+    out = xr.where(dac > base_doy, dac, dac + doy_max) - start_doy
+    out.attrs.update(da.attrs)
+    if start is not None:
+        out.attrs.update(units=f"days after {start}")
+    else:
+        out.attrs.update(units="days after time coordinate")
+
```

Comment:
```suggestion
out.name = da.name
```
Can confirm da.name is lost somewhere along the line